File: app/src/main/java/com/ekm/hairdo/things/Stack.java
package com.ekm.hairdo.things;

import java.io.Serializable;

public class Stack implements Serializable {

    public String address;
    public String url;

    public String price;

    public String lat, lng;
    public String name;

    public String hairid;
    public String styleName;

    public String stylistId;

    String distance;

    public boolean fav=false, favUpdated=false;

    public Stack() {
    }

    public Stack(String url) {
        this.url = url;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getPrice() {
        return price;
    }

    public void setPrice(String price) {
        this.price = price;
    }

    public String getLat() {
        return lat;
    }

    public void setLat(String lat) {
        this.lat = lat;
    }

    public String getLng() {
        return lng;
    }

    public void setLng(String lng) {
        this.lng = lng;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getHairid() {
        return hairid;
    }

    public void setHairid(String hairid) {
        this.hairid = hairid;
    }

    public String getStyleName() {
        return styleName;
    }

    public void setStyleName(String styleName) {
        this.styleName = styleName;
    }

    public String getStylistId() {
        return stylistId;
    }

    public void setStylistId(String stylistId) {
        this.stylistId = stylistId;
    }

    public String getDistance() {
        return distance;
    }

    public void setDistance(String distance) {
        this.distance = distance;
    }


    public boolean isFav() {
        return fav;
    }

    public void setFav(boolean fav) {
        this.fav = fav;
    }

    public boolean isFavUpdated() {
        return favUpdated;
    }

    public void setFavUpdated(boolean favUpdated) {
        this.favUpdated = favUpdated;
    }
}

File: app/src/main/java/com/ekm/hairdo/things/Stack.kt
package com.ekm.hairdo.things

import android.os.Parcelable
import android.util.Log
import com.google.firebase.firestore.DocumentSnapshot
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Stack ( var address: String,
                   var url: String,
                   var price: String,
                   var lat: String,
                   var lng:String,
                   var name: String,
                   var hairid: String,
                   var styleName: String,
                   var stylistId: String,
                   var distance: String,
                   var fav: Boolean,
                   var favUpdated:Boolean) : Parcelable {

    companion object {
        fun DocumentSnapshot.toStack(): Stack? {
            try {

                var address = getString("address")!!
                var url = getString("url")!!
                var price = getString("price")!!
                var lat = getString("lat")!!
                var lng:String = getString("lng")!!
                var name = getString("name")!!
                var hairid = getString("hairid")!!
                var styleName = getString("styleName")!!
                var stylistId = getString("stylistId")!!
                var distance = getString("distance")!!
                var fav: Boolean = getBoolean("fav")!!
                var favUpdated:Boolean = getBoolean("favUpdated")!!
                return Stack(address,url,price,lat,lng,name,hairid,styleName,stylistId,distance,fav,favUpdated)
            } catch (e: Exception) {
                Log.e(TAG, "Error converting user profile", e)
//                FirebaseCrashlytics.getInstance().log("Error converting user profile")
//                FirebaseCrashlytics.getInstance().setCustomKey("userId", id)
//                FirebaseCrashlytics.getInstance().recordException(e)
                return null
            }
        }
        private const val TAG = "Stack"
    }

}